spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/DeleteOrphanFilesSparkAction.java

```diff
@@ -326,6 +344,96 @@ private static FlatMapFunction<Iterator<String>, String> listDirsRecursively(
     };
   }
 
+  @VisibleForTesting
+  static List<String> findOrphanFiles(SparkSession spark,
+                                      Dataset<Row> actualFileDF,
+                                      Dataset<Row> validFileDF,
+                                      Map<String, String> equalSchemes,
+                                      Map<String, String> equalAuthorities,
+                                      PrefixMismatchMode prefixMismatchMode) {
+    Map<String, String> equalSchemesMap = flattenMap(equalSchemes);
+    Map<String, String> equalAuthoritiesMap = flattenMap(equalAuthorities);
+
+    Dataset<PathProxy> normalizedActualFileDF = actualFileDF.mapPartitions(
+        toFileMetadata(equalSchemesMap, equalAuthoritiesMap),
+        Encoders.bean(PathProxy.class)).as("actual");
+    Dataset<PathProxy> normalizedValidFileDF = validFileDF.mapPartitions(
+        toFileMetadata(equalSchemesMap, equalAuthoritiesMap),
+        Encoders.bean(PathProxy.class)).as("valid");
+
+    Column actualFileName = normalizedActualFileDF.col("path");
```

Comment:
I don't think the var names are correct anymore. What about having `joindCond` to shorten the line below?

```
Column joinCod = actualFileMetadataDS.col("path").equalTo(validFileMetadataDS.col("path"));

List<String> orphanFiles = actualFileMetadataDS
    .joinWith(validFileMetadataDS, joinCod, "leftouter")
    .mapPartitions(findOrphanFilesMapPartitions(prefixMismatchMode, conflicts), Encoders.STRING())
    .collectAsList();
```

Reply:
done